ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp

```diff
@@ -7176,6 +7176,50 @@ const char* GDALGeoPackageDataset::GetGeometryTypeString(OGRwkbGeometryType eTyp
     return pszGPKGGeomType;
 }
 
+/************************************************************************/
+/*                           GetFieldDomainNames()                      */
+/************************************************************************/
+
+std::vector<std::string> GDALGeoPackageDataset::GetFieldDomainNames(CSLConstList) const
+{
+    if( !HasDataColumnConstraintsTable() )
+        return std::vector<std::string>();
+
+    std::vector<std::string> oDomainNamesList;
+
+    std::unique_ptr<SQLResult> oResultTable;
+    {
+        std::string osSQL =
+            "SELECT constraint_name "
+            "FROM gpkg_data_column_constraints "
+            "WHERE constraint_name NOT LIKE '_%_domain_description' "
+            "ORDER BY constraint_name "
+            "LIMIT 10000" // to avoid denial of service
+            ;
+        oResultTable = SQLQuery(hDB, osSQL.c_str());
+        if( !oResultTable )
+            return oDomainNamesList;
+    }
+
+    if( oResultTable->RowCount() == 10000 )
+    {
+        CPLError(CE_Warning, CPLE_AppDefined,
+                 "Number of rows returned for field domain names has been "
+                 "truncated.");
+    }
+    else if( oResultTable->RowCount() > 0 )
+    {
+        oDomainNamesList.reserve(oResultTable->RowCount());
+        for ( int i = 0; i < oResultTable->RowCount(); i++ )
+        {
+            const char *pszConstraintName = oResultTable->GetValue(0, i);
```

Comment:
pszConstraintName should be tested against nullptr (even if the official schema from the GPKG spec would have a not null constraint, nothing guarantee that the database we open respect it)

Reply:
Fixed, thanks